src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEPropertySymbol.cs

```diff
@@ -64,6 +62,42 @@ private enum Flags : byte
             Debug.Assert((object)containingType != null);
             Debug.Assert(!handle.IsNil);
 
+            var metadataDecoder = new MetadataDecoder(moduleSymbol, containingType);
+            SignatureHeader callingConvention;
+            BadImageFormatException propEx;
+            var propertyParams = metadataDecoder.GetSignatureForProperty(handle, out callingConvention, out propEx);
+            Debug.Assert(propertyParams.Length > 0);
+
+            var returnInfo = propertyParams[0];
+            PEPropertySymbol result;
+
+            if (returnInfo.CustomModifiers.IsDefaultOrEmpty && returnInfo.RefCustomModifiers.IsDefaultOrEmpty)
```

Comment:
Consider changing `MetadataDecoder.GetSignatureForProperty` to return non-default `CustomModifiers` always.